java-security/src/test/java/com/sap/cloud/security/token/validation/validators/JwtIssuerValidatorTest.java

```diff
@@ -80,43 +71,38 @@ public void tokenIssuerWithoutHttpSchemeMatchesIdentityProviderUrl() {
-		assertThat(cut.validate(token).isValid(), is(true));
-	}
-
-	@Test
-	public void validationFails_whenTokenIssuerDoesNotMatchIdentityProviderUrl() {
-		cut = new JwtIssuerValidator(URI.create("https://paas.accounts400.ondemand.com"));
-		when(token.getClaimAsString(ISSUER)).thenReturn("https://accounts300.ondemand.com");
+	public void validationFails_whenIssuerEndsWithIdentityProviderUrlQueryParameter() {
+		cut = new JwtIssuerValidator(Collections.singletonList("accounts400.ondemand.com"));
+		when(token.getClaimAsString(ISSUER)).thenReturn("https://otherDomain.org?accounts400.ondemand.com");
 		ValidationResult validationResult = cut.validate(token);
 		assertThat(validationResult.isErroneous(), is(true));
 		assertThat(validationResult.getErrorDescription(), startsWith(
-				"Issuer is not trusted because 'iss' 'https://accounts300.ondemand.com' does not match one of these domains '[paas.accounts400.ondemand.com]' of the identity provider."));
+				"Issuer is not trusted because 'iss' claim 'https://otherDomain.org?accounts400.ondemand.com' does not provide a valid URI. Please contact your Identity Provider Administrator."));
 	}
 
 	@Test
-	public void validationFails_whenTokenIssuerEndsWithIdentityProviderUrlQueryParameter() {
+	public void validationFails_whenIasIssuerEndsWithIdentityProviderUrlQueryParameter() {
 		cut = new JwtIssuerValidator(Collections.singletonList("accounts400.ondemand.com"));
-		when(token.getClaimAsString(ISSUER)).thenReturn("https://otherDomain.org?accounts400.ondemand.com");
+		when(token.getClaimAsString(ISSUER)).thenReturn("https://test.accounts400.ondemand.com");
+		when(token.getClaimAsString(IAS_ISSUER)).thenReturn("https://otherDomain.org?accounts400.ondemand.com");
 		ValidationResult validationResult = cut.validate(token);
 		assertThat(validationResult.isErroneous(), is(true));
 		assertThat(validationResult.getErrorDescription(), startsWith(
-				"Issuer is not trusted because 'iss' 'https://otherDomain.org?accounts400.ondemand.com' does not match one of these domains '[accounts400.ondemand.com]' of the identity provider."));
+				"Issuer is not trusted because 'ias_iss' claim 'https://otherDomain.org?accounts400.ondemand.com' does not provide a valid URI."));
 	}
 
 	@Test
-	public void validationFails_whenTokenIssuerIsEmpty() {
+	public void validationFails_whenIssuerIsEmpty() {
 		when(token.getClaimAsString(ISSUER)).thenReturn(" ");
+		when(token.getClaimAsString(IAS_ISSUER)).thenReturn("https://test.accounts400.ondemand.com");
 		ValidationResult validationResult = cut.validate(token);
 		assertThat(validationResult.isErroneous(), is(true));
 		assertThat(validationResult.getErrorDescription(),
 				startsWith("Issuer validation can not be performed because Jwt token does not contain 'iss' claim."));
 	}
 
 	@Test
-	public void validationFails_whenTokenIssuerIsNull() {
+	public void validationFails_whenIssuerIsNull() {
```

Comment:
just a suggestion, maybe use Parameterized Tests here, seems that you have a pattern for tests that could be parameterized like [here](https://github.com/SAP/cloud-security-xsuaa-integration/blob/9713812ddbc8cb353188725b9067a651eb252952/spring-xsuaa/src/test/java/com/sap/cloud/security/xsuaa/token/XsuaaTokenTest.java#L100)?